starter/src/index.js

```diff
@@ -0,0 +1,109 @@
+import Apify from 'apify';
+import { inspect } from 'util';
+import { convertInputToActorConfigs } from './lib/configs.js';
+import { waitForRunToFinishAndPushData, startRun } from './lib/startRunAndPool.js';
+
+const { log } = Apify.utils;
+const env = Apify.getEnv();
+
+Apify.main(async () => {
+    /** @type {import('../../common/types').ActorInputData} */
+    // @ts-ignore It's not null
+    const input = await Apify.getInput();
+    log.debug('Provided inputs:');
+    log.debug(inspect(input));
+
+    const { maxConcurrentDomainsChecked, urlsToCheck } = input;
+
+    // Log the input
+    log.info('Input provided:');
+    log.debug(inspect(input, false, 4));
+
+    /** @type {import('./types').FrontendActorState} */
+    // @ts-expect-error It's an object
+    const state = await Apify.getValue('STATE') ?? {
+        runConfigurations: [],
+        totalUrls: urlsToCheck.length,
+        checkerFinished: false,
+    };
+
+    Apify.events.on('migrating', async () => {
+        await Apify.setValue('STATE', state);
+    });
+
+    Apify.events.on('persistState', async () => {
+        await Apify.setValue('STATE', state);
+    });
+
+    setInterval(async () => {
+        await Apify.setValue('STATE', state);
+
+        log.debug('Internal state:');
+        log.debug(inspect(state, false, 3));
+    }, 10_000);
```

Comment:
'migrating' event is part of 'persistState' and I think we can remove the `Apify.setValue` from setInterval too since we tell what happened to the user anyway by pushing the results to dataset

Reply:
Hmm, good point, the internal state is just run configurations, so there's nothing to debug. Will remove this interval and the migrating event handler 👍